src/github.com/matrix-org/dendrite/roomserver/storage/storage.go

```diff
@@ -104,15 +106,15 @@ func (d *Database) StoreEvent(event gomatrixserverlib.Event, authEventNIDs []typ
 	}, nil
 }
 
-func (d *Database) assignRoomNID(roomID string) (types.RoomNID, error) {
+func (d *Database) assignRoomNID(txn *sql.Tx, roomID string) (types.RoomNID, error) {
```

Comment:
I guess I don't mind, provided *all* the functions that take a `*sql.Tx` handle it being `nil`. The thing that would really annoy me is if I have to go and chase through 15 layers of function calls to check if I need to fix anything before passing `nil`.